.github/workflows/github-pages.yml

```diff
@@ -0,0 +1,53 @@
+name: Publish
+
+on:
+  push:
+    branches:
+    - main
+  # Allow workflow to be called manually
+  workflow_dispatch:
+
+jobs:
+  build-and-upload:
+    permissions:
+      contents: read
+      pages: read
```

Comment:
Maybe this needs to be write? https://github.com/nodejs/nodejs.dev/runs/8229047784?check_suite_focus=true

Reply:
I think we might need to enable pages from actions in the settings menu for this to work. I have these exact same settings working in another repo